app/src/main/java/fr/free/nrw/commons/mwapi/NetworkInterceptors.java

```diff
@@ -0,0 +1,85 @@
+package fr.free.nrw.commons.mwapi;
+
+import android.support.annotation.NonNull;
+
+import org.apache.http.Header;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.impl.client.ClientParamsStack;
+import org.apache.http.params.HttpParamsNames;
+import org.apache.http.protocol.HttpContext;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import timber.log.Timber;
+
+public class NetworkInterceptors {
+
+    /**
+     * Interceptor to log the HTTP request
+     */
+    @NonNull
+    public static HttpRequestInterceptor getHttpRequestInterceptor() {
+        return (HttpRequest request, HttpContext httpContext) -> {
```

Comment:
To minimize impact on performance, how about this:
- Check whether the app in in prod or not
- If prod, do nothing. If not, do the logging
? In the spirit of rule 4 at https://gualtierotesta.wordpress.com/2015/11/29/the-5-java-logging-rules/
Or we can leave it on for everybody until we don't have anymore network related bugs, and maybe do that check from 2.9.

Reply:
Yes, for now, we can probably log both for debug and release builds. Later on, we can consider running it just on debug builds.